cmd/gardener-extension-validator-gcp/app/app.go

```diff
@@ -0,0 +1,90 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app
+
+import (
+	"context"
+	"fmt"
+
+	gcpinstall "github.com/gardener/gardener-extension-provider-gcp/pkg/apis/gcp/install"
+	providergcp "github.com/gardener/gardener-extension-provider-gcp/pkg/gcp"
+	"github.com/gardener/gardener-extension-provider-gcp/pkg/validator"
+	controllercmd "github.com/gardener/gardener-extensions/pkg/controller/cmd"
+	"github.com/gardener/gardener-extensions/pkg/util"
+	"github.com/gardener/gardener/pkg/apis/core/install"
+
+	"github.com/spf13/cobra"
+	componentbaseconfig "k8s.io/component-base/config"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+var log = logf.Log.WithName("garden-extension-validator-gcp")
+
+// NewValidatorCommand creates a new command for running an GCP garden-extension-validator-gcp.
+func NewValidatorCommand(ctx context.Context) *cobra.Command {
+	var (
+		restOpts = &controllercmd.RESTOptions{}
+		mgrOpts  = &controllercmd.ManagerOptions{
+			WebhookServerPort: 443,
+		}
+
+		aggOption = controllercmd.NewOptionAggregator(
+			restOpts,
+			mgrOpts,
+		)
+	)
+
+	cmd := &cobra.Command{
+		Use: fmt.Sprintf("garden-extension-validator-%s", providergcp.Type),
+
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := aggOption.Complete(); err != nil {
+				controllercmd.LogErrAndExit(err, "Error completing options")
+			}
+
+			util.ApplyClientConnectionConfigurationToRESTConfig(&componentbaseconfig.ClientConnectionConfiguration{
+				QPS:   100.0,
+				Burst: 130,
+			}, restOpts.Completed().Config)
+
+			mgr, err := manager.New(restOpts.Completed().Config, mgrOpts.Completed().Options())
+			if err != nil {
+				controllercmd.LogErrAndExit(err, "Could not instantiate manager")
+			}
+
+			install.Install(mgr.GetScheme())
+
+			if err := gcpinstall.AddToScheme(mgr.GetScheme()); err != nil {
+				controllercmd.LogErrAndExit(err, "Could not update manager scheme")
+			}
+
+			log.Info("Setting up webhook server")
+			hookServer := mgr.GetWebhookServer()
+
+			log.Info("Registering webhooks")
+			hookServer.Register("/webhooks/validate-shoot", &webhook.Admission{Handler: &validator.Shoot{Logger: log.WithName("shoot-garden-extension-validator-gcp")}})
```

Comment:
```suggestion
			hookServer.Register("/webhooks/validation", &webhook.Admission{Handler: &validator.Shoot{Logger: log.WithName("shoot-garden-extension-validator-gcp")}})
```

Let's switch to a generic endpoint like `validation` as validation logic for further resources, e.g. `CloudProfile` will be added in the future. We can then build an internal logic (in another PR) which dispatches to the different validations. /cc @rfranzke